clamav_rest.py

```
@@ -192,6 +205,62 @@ def request_entity_too_large(error):
     return "File Too Large", 413
 
 
+@app.after_request
+def after_request(response):
+    """ Logging after every request. """
+    zipkin_headers = ("X-B3-Traceid", "X-B3-Spanid")
+    extra_labels = {"X-B3-Traceid": "none", "X-B3-Spanid": "none"}
+    for header in request.headers:
+        if header[0] in zipkin_headers:
+            extra_labels[header[0]] = header[1]
+    labels={
+        "trace.id": extra_labels["X-B3-Traceid"],
+        "span.id": extra_labels["X-B3-Spanid"]
+    }
+    try:
+        labels={**labels,
+            "http.request.body.content": request.data,
+            "http.request.body.bytes": len(request.data)
+        }
+    except:
```

a blind except is not good python because it could silence errors that you actually want to be aware of.

It's better to specify the exceptions that you want to catch, so any unexpected errors are allowed to propagate

```
try:
    # stuff
except AttributeError:
   .. 
```

However, it looks like what you're doing isn't needed. You could replace the whole thing with:

```
labels["http.request.body.content"] = getattr(request, "data", None)
labels["http.request.body.bytes"] = len(getattr(request, "data", []))
```

---

Thanks.
Yep, I've updated with that - though it still throws errors when the "413 Request Entity Too Large" error hits. The `getattr(request, "data", None)` line I think would work if `data` was absent - but when a 413 hits then even trying to read `request` seems problematic.
If there's an exception type that represents "Request Entity Too Large", I couldn't spot it.
See what you think of the update though.